pkg/registry/core/persistentvolume/strategy_test.go

```diff
@@ -34,3 +41,274 @@ func TestSelectableFieldLabelConversions(t *testing.T) {
 		map[string]string{"name": "metadata.name"},
 	)
 }
+
+func TestStatusUpdate(t *testing.T) {
+	now := metav1.Now()
+	origin := metav1.NewTime(now.Add(time.Hour))
+	later := metav1.NewTime(now.Add(time.Hour * 2))
+	nowFunc = func() metav1.Time { return now }
+	defer func() {
+		nowFunc = metav1.Now
+	}()
+	tests := []struct {
+		name        string
+		fg          bool
+		oldObj      *api.PersistentVolume
+		newObj      *api.PersistentVolume
+		expectedObj *api.PersistentVolume
+	}{
+		{
+			name: "feature enabled: timestamp is updated when phase changes",
+			fg:   true,
+			oldObj: &api.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Status: api.PersistentVolumeStatus{
+					Phase: api.VolumePending,
+				},
+			},
+			newObj: &api.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Status: api.PersistentVolumeStatus{
+					Phase: api.VolumeBound,
+				},
+			},
+			expectedObj: &api.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Status: api.PersistentVolumeStatus{
+					Phase:                   api.VolumeBound,
+					LastPhaseTransitionTime: &now,
+				},
+			},
+		},
+		{
+			name: "feature enabled: user timestamp change is respected on no phase change",
+			fg:   true,
+			oldObj: &api.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Status: api.PersistentVolumeStatus{
+					Phase: api.VolumePending,
+				},
+			},
+			newObj: &api.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Status: api.PersistentVolumeStatus{
+					Phase:                   api.VolumePending,
+					LastPhaseTransitionTime: &later,
+				},
+			},
+			expectedObj: &api.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+				Status: api.PersistentVolumeStatus{
+					Phase:                   api.VolumePending,
+					LastPhaseTransitionTime: &later,
+				},
+			},
+		},
+		{
+			name: "feature enabled: user timestamp is respected on phase change",
```

Comment:
Can you add one more test case for the timestamp being updated on phase change but the user didn't update the timestamps?

Reply:
@msau42 Added.

Reply:
I didn't see the new test case. I am thinking of the case where both old.LastPhaseTransitionTime and new.LastPhaseTransitionTime are the same, but the phase changes.

Reply:
Ack, that's a different test case, adding one more.